libs/features/creature/src/creature-text/creature-text.integration.spec.ts

```diff
@@ -0,0 +1,280 @@
+        'DELETE FROM `creature_text` WHERE (`CreatureID` = 1234) AND (`ID` IN (0));\n' +
+          'INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES\n' +
+          "(1234, 0, 0, 'newText', 0, 0, 1, 0, 0, 0, 0, 0, '');",
+      );
+      page.expectFullQueryToContain(
+        'DELETE FROM `creature_text` WHERE (`CreatureID` = 1234);\n' +
+          'INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES\n' +
+          "(1234, 0, 0, 'newText', 0, 0, 1, 0, 0, 0, 0, 0, '');",
+      );
+
+      page.setInputValueById('GroupID', '123');
+      page.expectDiffQueryToContain(
+        'DELETE FROM `creature_text` WHERE (`CreatureID` = 1234) AND (`ID` IN (0));\n' +
+          'INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES\n' +
+          "(1234, 123, 0, 'newText', 0, 0, 1, 0, 0, 0, 0, 0, '');",
+      );
+      page.expectFullQueryToContain(
+        'DELETE FROM `creature_text` WHERE (`CreatureID` = 1234);\n' +
+          'INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES\n' +
+          "(1234, 123, 0, 'newText', 0, 0, 1, 0, 0, 0, 0, 0, '');",
+      );
+    });
+
+    it('adding a row, changing its values and duplicating it should correctly update the queries', () => {
+      page.addNewRow();
+      page.setInputValueById('Probability', '1');
+      page.setInputValueById('Text', 'newText');
+      page.setInputValueById('GroupID', '123');
+      page.duplicateSelectedRow();
+
+      page.expectDiffQueryToContain(
+        'DELETE FROM `creature_text` WHERE (`CreatureID` = 1234) AND (`ID` IN (0, 1));\n' +
+          'INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES\n' +
+          "(1234, 123, 0, 'newText', 0, 0, 1, 0, 0, 0, 0, 0, ''),\n" +
+          "(1234, 123, 1, 'newText', 0, 0, 1, 0, 0, 0, 0, 0, '');",
+      );
+      page.expectFullQueryToContain(
+        'DELETE FROM `creature_text` WHERE (`CreatureID` = 1234);\n' +
+          'INSERT INTO `creature_text` (`CreatureID`, `GroupID`, `ID`, `Text`, `Type`, `Language`, `Probability`, `Emote`, `Duration`, `Sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES\n' +
+          "(1234, 123, 0, 'newText', 0, 0, 1, 0, 0, 0, 0, 0, ''),\n" +
+          "(1234, 123, 1, 'newText', 0, 0, 1, 0, 0, 0, 0, 0, '');",
+      );
+    });
+  });
+
+  describe('Editing existing', () => {
+    beforeEach(() => setup(false));
```

Comment:
this `beforeEach` will become obsolete with the new `setup` structure suggested in my comment above